avro-serializer/src/main/java/io/confluent/kafka/serializers/AbstractKafkaAvroSerializer.java

```diff
@@ -33,7 +33,7 @@
 public abstract class AbstractKafkaAvroSerializer extends AbstractKafkaAvroSerDe {
   private final EncoderFactory encoderFactory = EncoderFactory.get();
 
```

Comment:
Can enableAutoSchemaRegistration be a protected boolean in AbstractKafkaAvroSerializer, which is just set in the various subclasses during init/config etc?

Then the signature of serializeImpl can remain unchanged, for example.